docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md

```diff
@@ -105,27 +102,31 @@ props.setProperty(ReadWriteSplittingPlugin.READER_HOST_SELECTOR_STRATEGY.name, "
+### Session state
 
-- autocommit value
-- transaction isolation level
-
-All other session state attributes will be lost when switching connections between the writer/reader.
-
-If your SQL workflow depends on session state attributes that are not mentioned above, you will need to re-configure those attributes each time that you switch between the writer/reader.
+The plugin supports session state transfer when switching connection. All attributes mentioned in [Session State](../SessionState.md) are automatically transferred to a new connection. 
 
 
 ### Limitations when using Spring Boot/Framework
 
 #### @Transactional(readOnly = True)
 
 > [!WARNING]\
-> The use of read/write splitting with the annotation @Transactional(readOnly = True) is not recommended.
+> The use of read/write splitting with the annotation @Transactional(readOnly = True) is **only** recommended for configurations using an internal connection pool. Using the annotation with any other configurations will cause a significant performance degradation.
 
 When a method with this annotation is hit, Spring calls conn.setReadOnly(true), executes the method, and then calls setReadOnly(false) to restore the connection's initial readOnly value. Consequently, every time the method is called, the plugin switches to the reader, executes the method, and then switches back to the writer. Although the reader connection will be cached after the first setReadOnly call, there is still some overhead when switching between the cached writer/reader connections. This constant switching is not an ideal use of the plugin because it is frequently incurring this overhead. The suggested approach for this scenario is to avoid loading the read/write splitting plugin and instead use the writer cluster URL for your write operations and the reader cluster URL for your read operations. By doing this you avoid the overhead of constantly switching between connections while still spreading load across the database instances in your cluster. 
 
 #### Internal connection pools
 
-We recommend that you do not enable internal connection pools when using Spring. This is because Spring by default uses its own external connection pool. The use of both internal and external pools is not tested and may result in problematic behavior.
+We recommend that you explicitly disable internal connection pools when using Spring. You can check `spring.datasource.type` property to ensure that an external connection pooling is disabled. By default, Spring uses datasource auto-detection, and it may enable external connection pool. The use of both internal and external pools has not been tested and may result in problematic behaviour. A Spring application should enable internal connection pool only. That's a recommended configuration.
```

Comment:
Does this paragraph mean that internal connection pooling isn't recommended to use, but if they are used, users should disable Spring's external connection pooling? And that both internal and external connection pooling shouldn't be used together? If that's the case, how does this sound:
```suggestion
We recommend that you explicitly disable internal connection pools when using Spring. However, if you do use the driver's internal connection pooling, you can check the `spring.datasource.type` property to ensure that any external connection pooling is disabled. This is necessary because Spring uses datasource auto-detection by default, and it may enable external connection pooling. Using internal and external pools at the same time has not been tested and may result in problematic behaviour. The recommended configuration for a Spring application should either enable internal connection pooling or external connection pooling, but not both at once.
```